cmd/skopeo/signing.go

```diff
@@ -88,3 +89,179 @@ var standaloneVerifyCmd = cli.Command{
 	ArgsUsage: "MANIFEST DOCKER-REFERENCE KEY-FINGERPRINT SIGNATURE",
 	Action:    standaloneVerify,
 }
+
+func addSignature(context *cli.Context) error {
+	if len(context.Args()) != 2 {
+		return errors.New("Usage: skopeo add-signature reference key-fingerprint")
+	}
+	ref := context.Args()[0]
+	fingerprint := context.Args()[1]
+
+	imageRef, err := transports.ParseImageName(ref)
+	if err != nil {
+		return fmt.Errorf("Error parsing %s: %v", ref, err)
+	}
+	name := transports.ImageName(imageRef)
+
+	reference := context.String("reference")
+	if reference == "" {
+		signReference := imageRef.DockerReference()
+		if signReference == nil {
+			return fmt.Errorf("No reference associated with image %s: %v", ref, err)
+		}
+		reference = signReference.String()
+	} else {
+		signRef, err := imageRef.Transport().ParseReference(reference)
+		if err != nil {
+			return fmt.Errorf("Error parsing %s: %v", reference, err)
+		}
+		signReference := signRef.DockerReference()
```

Comment:
This won’t work; if `imageRef.DockerReference()` (in the `reference == ""` case) is `nil`, i.e. the user _needs_ to supply a reference, `signRef.DockerReference()` will most likely also be `nil` in this case (it will be calling the same implementation).  Either just use the input `reference` string as is, or, to validate it, parse it using`containers/image/docker/reference.ParseNamed` possibly with `.WithDefaultTag`.